tasks/mount.yml

```diff
@@ -0,0 +1,106 @@
+  fail:
+    msg: "Sorry, we only support CEPHFS right now."
+  when: os_manila_mount_share_protocol != "CEPHFS"
+
+- name: Install Ceph Repos
+  block:
+  - yum_repository:
+      name: ceph-noarch
+      description: Ceph - noarch
+      baseurl: "{{ os_manila_mount_ceph_repo_base }}/noarch"
+      gpgkey: "{{ os_manila_mount_ceph_repo_key | default(omit) }}"
+  - yum_repository:
+      name: ceph-arch
+      description: Ceph - arch
+      baseurl: "{{ os_manila_mount_ceph_repo_base }}/$basearch"
+      gpgkey: "{{ os_manila_mount_ceph_repo_key | default(omit) }}"
+  - yum: name=epel-release,ceph-common,ceph-fuse,fuse
+  when: "ansible_os_family == 'RedHat' and 
+         ansible_distribution_major_version == '7' and
+         os_manila_mount_pkgs_install"
+  become: yes
+
+- name: Ensure ceph conf directory exists
+  file:
+    path: "{{ os_manila_mount_ceph_conf_path }}"
+    state: directory
+    mode: 0755
+    owner: root
+    group: root
+  become: yes
+
+- name: Configure ceph.conf using os_manila_mount_host
+  template:
+    src: ceph.conf.j2
+    dest: "{{ os_manila_mount_ceph_conf_path }}/ceph.conf"
+    owner: root
+    group: root
+    mode: 0600
+  become: yes
+
+- name: Ensure mount directory exists
+  file:
+    path: "{{ os_manila_mount_path }}"
+    state: directory
+    owner: "{{ os_manila_mount_user }}"
+    group: "{{ os_manila_mount_group }}"
+    mode: "{{ os_manila_mount_mode }}"
+  become: yes
+
+
+# Using _netdev as a filesystem option prevents the mount from blocking early boot
+# before networking is initialised
+
+- block:
+  - name: Add user keyring with os_manila_mount_access_key
+    template:
+      src: keyring.j2
+      dest: "{{ os_manila_mount_ceph_conf_path }}/ceph.client.{{ os_manila_mount_share_user }}.keyring"
+      owner: root
+      group: root
+      mode: 0600
+  - name: FUSE-mount the share from the Ceph cluster
+    mount:
+      path: "{{ os_manila_mount_path }}"
+      src: "none"
+      fstype: fuse.ceph
+      state: mounted
+      opts: "_netdev,noatime,ceph.id={{ os_manila_mount_share_user }},ceph.conf={{ os_manila_mount_ceph_conf_path}}/ceph.conf,ceph.client-mountpoint={{ os_manila_mount_export }}"
+  when: os_manila_mount_fuse
+  become: yes
+
+- block: 
+  - name: Write Ceph client secret
+    copy:
+      dest: "{{ os_manila_mount_ceph_conf_path }}/ceph.client.{{ os_manila_mount_share_user }}.secret"
+      content: "{{ os_manila_mount_access_key }}"
+      mode: 0600
+      owner: root
+      group: root
+  - name: FS-Mount the share from the Ceph cluster
+    mount:
+      path: "{{ os_manila_mount_path }}"
+      src: "{{ os_manila_mount_host }}:{{ os_manila_mount_export }}"
+      fstype: ceph
+      state: mounted
+      opts: "_netdev,noatime,name={{os_manila_mount_share_user}},secretfile=/etc/ceph/ceph.client.{{os_manila_mount_share_user}}.secret"
+      boot: yes
+  when: not os_manila_mount_fuse
+  become: yes
+
+- name: Ensure mounted directory has correct permissions
```

Comment:
This is done on L49. Does it need to be done twice?

Reply:
So the second time is interesting, it actually changes permissions in mounted remote filesystem. Now I needed that originally to ensure on a fresh manila share the appropriate users could actually write files. Probably should be an optional step though, as it will overwrite the existing permissions that could be set on a share that has other things done to it.